### PR TITLE
CLI for `get_manifest` method

### PR DESCRIPTION
**Pull request by sujaypatil96**
Based on discussion in issue #328 and `click`-example, adding a CLI for the `get_manifest()` method.

To run this command:

`$ python -m schematic manifest get TITLE PATH_TO_JSON_LD DATA_TYPE --DATASET_ID --SHEET_URL --JSON_SCHEMA`

### CODE REVIEW

**Review by milen-sage — CHANGES_REQUESTED** [review]
Reflecting our conversation today: 
- provide handling of parameters specified via the config file 
- issue warning when cli positional/keyword parameters overlap with parameters provided in config (i.e. indicating that CLI-provided parameters override the config ones)

**Comment by BrunoGrandePhD** [comment]
I'll review this once the changes that were mentioned in our meeting today are implemented. Ping me once that's ready!

**Comment by sujaypatil96** [comment]
@milen-sage  / @BrunoGrandePhD: The code now uses parameter values from the `config.yml` file. 

With respect to raising warnings using `logging` module, I'll implement it when tackling issue #317. For now, it simply logs to `stdout` instead of `stderr`.

File: schematic/manifest/commands.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+def manifest(): # use as `schematic manifest ...`
+    """
+    Sub-commands with Manifest Generation utilities/methods.
+    """
+    pass
+
+
+# prototype based on getModelManifest() and get_manifest()
+# use as `schematic config get positional_args --optional_args`
+@manifest.command('get', short_help='Prepares the manifest URL based on provided schema.')
```
Comment (BrunoGrandePhD) [inline schematic/manifest/commands.py:20]: We might want to better distinguish between mandatory or optional arguments by switching some arguments to being positional.

**Review by BrunoGrandePhD — CHANGES_REQUESTED** [review]
I started doing my review, but then I started implementing some utility functions to minimize code duplication. See #368.

File: schematic/manifest/generator.py [hunk 2]
```diff
@@ -192,7 +193,7 @@ def _get_valid_values_from_jsonschema_property(self, prop:dict) -> List[str]:
             return []
 
 
-    def get_empty_manifest(self, json_schema = None):
+    def get_empty_manifest(self, json_schema_file = None):
```
[the remark this answers is not in the document]
Comment (sujaypatil96) [reply schematic/manifest/generator.py]: I agree. I've made a change to the `get_empty_manifest()` prototype, to change `json_schema_file` argument to `json_schema_filepath`.

File: schematic/manifest/commands.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+# define the optional arguments
+@click.option('-t', '--title', help='Title of generated manifest file.')
+@click.option('-d', '--data_type', help='Data type/component from JSON-LD schema to be used for manifest generation.')
+@click.option('-p', '--path_to_json_ld', help='Path to JSON-LD schema.')
```
[the remark this answers is not in the document]
Comment (sujaypatil96) [reply schematic/manifest/commands.py]: Sounds good. The `schematic manifest get` CLI now uses `--jsonld` as an optional argument.

File: schematic/manifest/commands.py (hunk 1 above)
Comment (sujaypatil96) [reply schematic/manifest/commands.py:23]: We agreed that we wouldn't have any positional arguments while defining our CLIs. All arguments will be optional arguments. To enforce certain arguments we would just create "required, optional" arguments.

File: schematic/utils/cli_utils.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            return None
+        return dictionary.get(key)
+
+    return reduce(extract, keys, dictionary)
+
+
+def fill_in_from_config(
+    arg_name: str,
+    arg_value: Any,
+    config_keys: Sequence[Any]
```
Comment (sujaypatil96) [inline schematic/utils/cli_utils.py:43]: Removed the `config` parameter that was previously being passed to `fill_in_from_config()`. Now it just uses the global `CONFIG` object.

File: schematic/manifest/commands.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+# use as `schematic config get positional_args --optional_args`
+@manifest.command('get', short_help='Prepares the manifest URL based on provided schema.')
+# define the optional arguments
+@click.option('-t', '--title', help='Title of generated manifest file.')
+@click.option('-d', '--data_type', help='Data type/component from JSON-LD schema to be used for manifest generation.')
+@click.option('-p', '--jsonld', help='Path to JSON-LD schema.')
+@click.option('-d', '--dataset_id', help='SynID of existing dataset on Synapse.')
+@click.option('-s', '--sheet_url', type=bool, help='Enable/disable URL generation.')
+@click.option('-j', '--json_schema', help='Path to JSON Schema (validation schema).')
+@click.option('-c', '--config', help='Path to schematic configuration file.', required=True)
```
Comment (sujaypatil96) [inline schematic/manifest/commands.py:31]: We can enforce `config` optional argument by setting `required=True`.

**Comment by sujaypatil96** [comment]
This PR is ready for review @BrunoGrandePhD.

Also, I checked the "bug" which was coming up when you were using `syn23643254` as the value for the `dataset_id` argument, and it is because there's no CSV (manifest) file in that folder. So it's a good idea to create a raise an informative exception in that case, and it will be a good idea to that in the `get_manifest()` method. I will save that bugfix for another PR though.

File: schematic/manifest/commands.py [hunk 6]
```diff
@@ -1,7 +1,10 @@
 #!/usr/bin/env python3
 
+import warnings
```
Comment (BrunoGrandePhD) [inline schematic/manifest/commands.py:3]: Why was this added? I don't see this module being used anywhere in this file.

**Review by BrunoGrandePhD — APPROVED** [review]
If you can remove the unused `warnings` import, go ahead and merge this PR! 

🚀

File: schematic/manifest/commands.py (hunk 6 above)
Comment (sujaypatil96) [reply schematic/manifest/commands.py]: Thanks, removing `warnings`.

**Review by milen-sage — APPROVED** [review]
